src/multivariate/dirichlet.jl

```diff
@@ -191,30 +191,30 @@ function suffstats(::Type{<:Dirichlet}, P::AbstractMatrix{Float64})
     K = size(P, 1)
     n = size(P, 2)
     slogp = zeros(K)
-    for i = 1:n
-        for k = 1:K
-            @inbounds slogp[k] += log(P[k,i])
+    for i in axes(P, 2)
+        for (k, kP) in zip(1:K, axes(P,1))
+            @inbounds slogp[k] += log(P[kP,i])
         end
     end
     DirichletStats(slogp, n)
 end
 
 function suffstats(::Type{<:Dirichlet}, P::AbstractMatrix{Float64},
-                   w::AbstractArray{Float64})
+                   w::AbstractArray{Float64,1})
```

Comment:
Why this change? This seems breaking?

Reply:
```suggestion
                   w::AbstractArray{Float64})
```

Reply:
The weights should be 1D array, shouldn't they? i.e. a vector with each element corresponding to a row in `P`? Or is there an example where it makes sense to have an array of arbitrary dimension ?

Reply:
Well, I don't know what people use. But the current implementation definitely allows to use higher-dimensional weight arrays (the code also only checks and works with `length(w)`), so any change here is breaking. I assume it's just similar to many functions in StatsBase that allow arrays for arbitrary and non-matching size as inputs if their lengths match.